relay-cardinality/src/redis/limiter.rs

```diff
@@ -162,12 +162,14 @@ impl Limiter for RedisSetLimiter {
             let id = &state.id().to_string();
             let scopes = num_scopes_tag(&state);
             let results = metric!(
-                timer(CardinalityLimiterTimers::Redis),
+                async_timer(CardinalityLimiterTimers::Redis),
                 id = id,
                 scopes = scopes,
-                { self.check_limits(&mut connection, &mut state, timestamp) }
-            )
-            .await?;
```

Comment:
Wasn't the problem just that you did not await inside the block? And there is no need to make a new macro?

Reply:
Maybe I overlooked something but we can't wrap an async function call with a non-async block, which is what happens in the macro.

Reply:
Am I crazy, this compiles for me:

```rust
            let results = metric!(
                timer(CardinalityLimiterTimers::Redis),
                id = id,
                scopes = scopes,
                {
                    self.check_limits(&mut connection, &mut state, timestamp)
                        .await
                }
            )?;
```

So does the minimal example:

```rust
async fn bar() {}

async fn foo() {
    let x = {
        bar().await;
    };
}
```

Reply:
I feel like I hallucinated, it works indeed, thanks!